hazelcast/src/main/java/com/hazelcast/client/impl/connection/tcp/TcpClientConnectionManager.java

```diff
@@ -422,9 +422,23 @@ private void submitConnectToClusterTask() {
                             logger.warning("No connection to cluster: " + clusterId);
                         }
 
-                        submitConnectToClusterTask();
+                        submitConnectToClusterTask(initialConnection);
```

Comment:
should we add a return after this? I believe under some scenarios, we might send the state/fire the lifecycle event twice now

Reply:
Changed it to a if/ else if instead